controllers/component_build_controller_simple_build.go

```diff
@@ -136,14 +135,18 @@ func (r *ComponentBuildReconciler) getBuildGitInfo(ctx context.Context, componen
 	}
 	if !isPublic {
 		// Try to find git secret in case private git repository is used
-		gitSecretName, err = r.findGitSecretName(ctx, component)
-		if err != nil {
-			log.Error(err, "failed to find git secret name")
-			return nil, err
-		}
+		gitSecretName = component.Spec.Secret
 		if gitSecretName == "" {
-			// The repository is private and no git secret provided
-			return nil, boerrors.NewBuildOpError(boerrors.EComponentGitSecretMissing, nil)
+			// The repository is private and no git secret specified
+			return nil, boerrors.NewBuildOpError(boerrors.EComponentGitSecretNotSpecified, nil)
```

Comment:
The diagram shows the process is:

UI HAC creates RemoteSecret -> RemoteSecret creates the Secret -> UI HAC update the Secret name to component

Should we treat the this error as transient error and emit Event as well?

Reply:
it might be both, but I'd stick to permanent, because user can edit Component spec manually.